### PR TITLE
feat: WASM module package signing [part 1]

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
# Description

Initial implementation of the WASM module package signing.
Added some basic structures and objects needed for package signing.

## Related Issue(s)

#249

## Description of Changes

- replaced `cryptoxide` crate usage with `RustCrypto` crates
- added `module sign` cli command
- added `Blake2b256` hashing object
- added ed25519 public and private keys objects with decoding from PEM format implementation
- added x.509 certificate object with decoding from PEM format implementation
- refactored some stuff

### CODE REVIEW

File: hermes/bin/src/runtime_extensions/hermes/crypto/bip39.rs [hunk 1]
```diff
@@ -168,20 +155,21 @@ fn generate_entropy(word_count: usize) -> Result<Vec<u8>, Errno> {
...
 
 /// Generate checksum bits from entropy bits.
+#[allow(clippy::indexing_slicing)]
 fn get_check_sum_bits(entropy_bits: &[u8], word_count: usize) -> Vec<u8> {
-    // Number of checksum bits to be added.
-    let check_sum_num = word_count / 3 * 4 * 8 / 32;
-    let mut hash_result = [0u8; 32];
-    let mut hasher = Sha256::new();
+    // Entropy bits number.
+    let entropy_bits_number = word_count * 4 * 8 / 3;
```
Comment (cong-or) [inline hermes/bin/src/runtime_extensions/hermes/crypto/bip39.rs:161]: where do these numbers come from?

File: hermes/bin/Cargo.toml [hunk 2]
```diff
@@ -63,12 +62,18 @@ console = { workspace = true }
 serde = { workspace = true, features = ["derive"] }
 serde_json = { workspace = true }
 jsonschema = { workspace = true }
+hex = { workspace = true }
+hmac = { workspace = true }
+pbkdf2 = { workspace = true }
+sha2 = { workspace = true }
+ed25519 = { workspace = true }
```
Comment (cong-or) [inline hermes/bin/Cargo.toml:69]: should ed25519-dalek be sufficient ? What's in ed25519?

File: hermes/bin/src/sign/certificate.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use temp_dir::TempDir;
+
+    use super::*;
+
+    #[test]
+    fn certificate_from_file_test() {
```
Comment (cong-or) [inline hermes/bin/src/sign/certificate.rs:73]: did you test parity between cryptoxide output and outputs after the introduction of dalek libs?

File: hermes/bin/Cargo.toml (hunk 2 above)
Comment (Mr-Leshiy) [reply hermes/bin/Cargo.toml:69]: Yea, good catch, `ed25519` unused dep, forgot to remove it

**Review by cong-or — APPROVED** [review]
lgtm

**Pull request merged**